sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinExec.scala

```diff
@@ -161,6 +164,10 @@ case class StreamingSymmetricHashJoinExec(
     new SerializableConfiguration(SessionState.newHadoopConf(
       sparkContext.hadoopConfiguration, sqlContext.conf)))
 
+
+  val nullLeft = new GenericInternalRow(left.output.map(_.withNullability(true)).length)
+  val nullRight = new GenericInternalRow(right.output.map(_.withNullability(true)).length)
```

Comment:
This can also be moved into the Joiner. Use `Joiner.inputAttributes`, instead of `left.output`/`right.output`
Basically if you have to define variables outside the Joine with the name "left", then it highly likely that you can move into the Joiner, as it is meant capture all the context needed to process one side.

Reply:
The problem is that the left joiner has left input attributes, but needs the right null row.

Reply:
right!! 
you could pass it on as another param `otherSideNullRow` in `storeAndJoinWithOtherSide`. I am fine either way.